src/posix/main.c

```diff
@@ -193,10 +193,9 @@ static otInstance *InitInstance(int aArgCount, char *aArgVector[])
 
     ParseArg(aArgCount, aArgVector, &config);
 
-#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED
```

Comment:
Why remove the line 196? Only the function "otPlatLog()" calls the function "syslog()".  If 'OPENTHREAD_CONFIG_LOG_OUTPUT' is set to other value. The function "otPlatLog()" won't be implemented and no one calls the function "syslog()". Then we don't need to call the function "openlog()" to initialize the  syslog.

Reply:
Actually, OpenThread's core configurations is not exposed as a user header file, so the posix-host assumes it's configured as platform defined way for logging. Actually, it also assumes dynamic logging is enabled.